src/merlin.ml

```diff
@@ -37,8 +37,13 @@ module Preprocess = struct
       No_preprocessing
     | Pps { loc ; pps = pps1; flags = flags1; staged = s1 },
       Pps { loc = _; pps = pps2; flags = flags2; staged = s2 } ->
+      let compare_flags f1 f2 =
+        String.compare
+          (Expander.expand_str expander f1)
+          (Expander.expand_str expander f2)
```

Comment:
It seems that we are going to do the expansion several times in this file. We should either do the expansion in `make`, or simply comapre the string unexpanded here. The latter seems simpler and enough.